### DISCUSSION TITLE
[AppService] statyczne adresy wychodzące

### DISCUSSION THREAD

**Discussion by mgrabarz** [AzureCommunityPL/group#113] [Ogólna]
Nie wiem jak Wam, ale mi umknął fakt, że AppService w wersji PxV3 [!!CHYBA!!] ma statyczne adresy wychodzące. Dotychczas żeby coś takiego osiągnąć potrzebne było ASE, a tu taka niespodzianka.

Zmiana adresu wychodzącego może nastąpić w następujących [przypadkach](https://docs.microsoft.com/en-us/azure/app-service/overview-inbound-outbound-ips#when-outbound-ips-change). Na liście nie ma PxV3, podobnie zresztą wyświetlana jest informacja w Portalu:
- Dla tańszych planów "Outgoing IP addresses for your app might change."
- Dla PxV3 takiego ostrzeżenia brak :)

**Comment by smereczynski** [comment]
A sprawdzałeś czy to nie błąd w opisie? :D

**Reply by mgrabarz** [reply]
Twardego sprawdzenia jeszcze nie robiłem. Jestem mocno przekonany, że to jednak NIE jest błąd. :)

**Reply by mgrabarz** [reply]
@smereczynski jednak zaczynam wątpić.

ALL - ktoś ma ochotę potestować?

**Reply by kamil-mrzyglod** [reply]
@mgrabarz Tak patrzę i nie bardzo widzę różnicę pomiędzy PxV3 a innymi planami - generalnie IP dostaję inne co jakiś czas :)

**Reply by mgrabarz** [reply]
Co rozumiesz przez inne IP? Robisz na tym PV3 scale out/over i to zmienia IP?

**Reply by kamil-mrzyglod** [reply]
Patrzyłem tylko na outbound IP przy trzecim scenariuszu z docsów czyli zmiana tier. Jest szansa aby przy scale out IP się zmieniło przy App Service?

**Reply by mgrabarz** [reply]
Spytałem u źródła ;) 
https://twitter.com/marekgrabarz/status/1316358080481964032?s=20

**Reply by mgrabarz** [reply]
> Patrzyłem tylko na outbound IP przy trzecim scenariuszu z docsów czyli zmiana tier. Jest szansa aby przy scale out IP się zmieniło przy App Service?

Akurat między tier to wiadomo. 
Chodzi o scenariusz, że wgrywasz w korpie apkę na powiedzmy P1v3 i masz gwarancję, że jeśli przyspiesz statycznie, to IPki zostaną niezmienione. To z kolei oznacza, że możesz te IPki wsadzić na whitelisty w systemach do których ten AppService dzwoni - a często taki właśnie jest wymóg i często to wymuszało użycie ASE.

**Reply by kamil-mrzyglod** [reply]
@mgrabarz Pomiędzy V2 czy V3 też przestawiałem i zmian na IP tutaj nie widziałem, ale osobiście daleki bym był by stwierdzić, że to miarodajny test(szczególnie że zmiana z P1 na S1 też potrafiła mi zachować outbound IP).

Z drugiej strony zawsze rozumiałem to w ten sposób, że outbound IP są współdzielone tak długo jak zachowujesz tier(scale unit) + region. Dokumentacja poszerza to jeszcze o odtworzenie zasobu w ramach innej RG(co jest w sumie dość oczywiste). 

Gdy zadamy pytanie "Co muszę zrobić, aby whitelisting App Service zadziałał?" i damy odpowiedź:
* nie możesz zmieniać tier
* nie możesz robić deploymentu do innej RG
* nie możesz zmienić regionu

To nie wyczerpuje to tematu? Rozumiem scenariusz "statyczności" ASE i jak zakładam, tutaj rozchodzi się o brak potwierdzenia, że PxV3 rządzi się takimi samymi prawami jak B / S / P?

**Reply by mgrabarz** [reply]
No to okazuje się że to bug. Pv3 nie różni się od Pv2 w tym względzie.

Tak jak @kamil-mrzyglod pisałeś, adresy  się nie zmienia jeśli nie będziesz kombinował z usuwaniem i tierami. Nie da się natomiast podpiąć statycznego IPka (jak w ASE lub jak w IP SSL).

**Comment by piotrgwiazda** [comment]
A to nie chodzi przypadkiem o
https://docs.microsoft.com/pl-pl/azure/app-service/web-sites-integrate-with-vnet
Będzie kierował ruch wychodzący przez vNet z flagą WEBSITE_VNET_ROUTE_ALL

śr., 14 paź 2020, 14:50 użytkownik Marek Grabarz <notifications@github.com>
napisał:

**Reply by kaluzaaa** [reply]
Nie

**Reply by mgrabarz** [reply]
To routuje do VNet ale dalej jakoś trzeba wyjść ze stałego adresu na swiat...
- NatGateway nie jest obecnie uwzględniany przez AppService jak zapniesz na subnet.
- Z Azure Firewall lub NVA pewnie się uda ale wyjdzie kilka razy drożej niż AppService

**Reply by miqm** [reply]
@mgrabarz - wydaje mi się, że NatGateway jest już uwzględniany przez AppService po zapięciu na subnet.

Mamy case, że NAT-GW był zapięty na subnet ale bez publicznego IP i do tej ruch wyjściowy działał, ale dzisiaj przestał. Po podpięciu IPka do NAT-GW zaczęło śmigać a sprawdzając publiczny adres ip z kontenera (request na ipfy.org) był to adres przypięty do natGW. 

Zatem wygląda, że to jednak działa, chociaż to będzie dotyczyć ruchu z aplikacji (kontenera z aplikacją) a nie z samego AppService, np. przy pobieraniu kontenerów.

**Reply by mgrabarz** [reply]
Znam Wasz przypadek i chyba wreszcie to zakodowali (jakoś w weekend poszło).

Można dopytać Krystynę... ;)
https://twitter.com/marekgrabarz/status/1314172163482624000?s=19

**Reply by miqm** [reply]
Jak rozumiem oprócz zagadania na twitterze nie ma jakiegoś publicznego change-loga? Przydałoby się aby móc wyjaśnić, co się stało że się zes..psuło :)

**Reply by mgrabarz** [reply]
Krystyna odpisała. 
Zmiana roluje się na całym świecie ale jeszcze nie weszła wszędzie. Czekają z announcement do momentu aż się deploy skończy.

Nie popsuło się. Zadziałało.
